### ISSUE TITLE
Can no longer compile fc 0.015 on Linux (Ubuntu 20.04)

### ISSUE THREAD

**Issue by rohel01** [romnn/cargo-feature-combinations#2]
Hello,

I am setting up a new Gitlab CI slave at work, but can no longer compile  cargo-feature-combinations v0.0.15 on Linux.
Error is:

```
❯ cargo install cargo-feature-combinations
    Updating crates.io index
  Installing cargo-feature-combinations v0.0.15
    Updating crates.io index
   Compiling proc-macro2 v1.0.69
   Compiling unicode-ident v1.0.12
   Compiling cfg-if v1.0.0
   Compiling autocfg v1.1.0
   Compiling libc v0.2.150
   Compiling serde v1.0.192
   Compiling once_cell v1.18.0
   Compiling memchr v2.6.4
   Compiling crossbeam-utils v0.8.16
   Compiling lazy_static v1.4.0
   Compiling scopeguard v1.2.0
   Compiling tracing-core v0.1.32
   Compiling thread_local v1.1.7
   Compiling memoffset v0.9.0
   Compiling crossbeam-epoch v0.9.15
   Compiling aho-corasick v1.1.2
   Compiling sharded-slab v0.1.7
   Compiling serde_json v1.0.108
   Compiling quote v1.0.33
   Compiling cc v1.0.83
   Compiling syn v2.0.39
   Compiling pin-project-lite v0.2.13
   Compiling equivalent v1.0.1
   Compiling hashbrown v0.14.2
   Compiling regex-syntax v0.8.2
   Compiling rayon-core v1.12.0
   Compiling tracing-subscriber v0.3.17
   Compiling crossbeam-deque v0.8.3
   Compiling tracing v0.1.40
   Compiling indexmap v2.1.0
   Compiling adler v1.0.2
   Compiling bit-vec v0.6.3
   Compiling thiserror v1.0.50
   Compiling backtrace v0.3.69
   Compiling gimli v0.28.0
   Compiling eyre v0.6.8
   Compiling semver v1.0.20
   Compiling ryu v1.0.15
   Compiling either v1.9.0
   Compiling camino v1.1.6
   Compiling itoa v1.0.9
   Compiling tracing-error v0.2.0
   Compiling regex-automata v0.4.3
   Compiling bit-set v0.5.3
   Compiling miniz_oxide v0.7.1
   Compiling serde_derive v1.0.192
   Compiling thiserror-impl v1.0.50
   Compiling regex v1.10.2
   Compiling addr2line v0.21.0
   Compiling vte_generate_state_changes v0.1.1
   Compiling object v0.32.1
   Compiling rustc-demangle v0.1.23
   Compiling owo-colors v3.5.0
   Compiling indenter v0.3.3
   Compiling utf8parse v0.2.1
   Compiling vte v0.11.1
   Compiling color-spantrace v0.2.0
   Compiling fancy-regex v0.11.0
   Compiling rayon v1.8.0
   Compiling strip-ansi-escapes v0.2.0
   Compiling itertools v0.11.0
   Compiling color-eyre v0.6.2
   Compiling termcolor v1.3.0
   Compiling cargo-platform v0.1.4
   Compiling serde_json_merge v0.0.4
   Compiling cargo_metadata v0.18.1
   Compiling cargo-feature-combinations v0.0.15
error[E0599]: `Vec<u8>` is not an iterator
   --> /home/melchiore/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-feature-combinations-0.0.15/src/lib.rs:436:18
    |
435 |               let output = strip_ansi_escapes::strip(colored_output.get_ref())
    |  __________________________-
436 | |                 .map(|out| String::from_utf8_lossy(&out).into_owned());
    | |                 -^^^ `Vec<u8>` is not an iterator; try calling `.into_iter()` or `.iter()`
    | |_________________|
    |
    |
   ::: /home/melchiore/.rustup/toolchains/nightly-2023-06-23-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:396:1
    |
396 |   pub struct Vec<T, #[unstable(feature = "allocator_api", issue = "32838")] A: Allocator = Global> {
    |   ------------------------------------------------------------------------------------------------ doesn't satisfy `Vec<u8>: Iterator`
    |
    = note: the following trait bounds were not satisfied:
            `Vec<u8>: Iterator`
            which is required by `&mut Vec<u8>: Iterator`
            `[u8]: Iterator`
            which is required by `&mut [u8]: Iterator`

error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
   --> /home/melchiore/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-feature-combinations-0.0.15/src/lib.rs:436:23
    |
436 |                 .map(|out| String::from_utf8_lossy(&out).into_owned());
    |                       ^^^ doesn't have a size known at compile-time
    |
    = help: the trait `Sized` is not implemented for `[u8]`
    = note: all function arguments must have a statically known size
    = help: unsized fn params are gated as an unstable feature

Some errors have detailed explanations: E0277, E0599.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `cargo-feature-combinations` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error: failed to compile `cargo-feature-combinations v0.0.15`, intermediate artifacts can be found at `/tmp/cargo-installJCXuVK`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

I tried with various Rust versions (stable and nightly). Last time I successfully installed it, I was using `nightly-2023-06-23`. Today, using the exact same version, compilation fails.

Any help would be appreciated.

**Comment by rohel01** [comment]
Crate is breaking because the `strip_ansi_escapes::strip` function return type has changed between versions 0.1 and 0.2 of the corresponding crates. cargo install does not have access to the `Cargo.lock` file from this repository, so it does not honor it.

**Comment by wez** [comment]
I just ran into this as well; you can `cargo install cargo-feature-combinations --locked` to get a successful compile

**Comment by wez** [comment]
Note that this issue is a duplicate of #1

**Comment by romnn** [comment]
Thanks for reporting the issue.

This problem is fixed by e5ff5ec and released in v0.0.16.

Closing this and #1.
